sdk/test-utils/recorder/src/utils.ts

```diff
@@ -21,17 +21,73 @@ export function isPlaybackMode() {
   return !isRecordMode() && !isLiveMode();
 }
 
+/**
+ * Encodes a string as a URI component, but also taking in consideration the RFC 3986 specification.
+ * JavaScript's encodeURIComponent method doesn't take in consideration the characters: !, ', (, ), *
+ * @param str The string that needs to be encoded.
+ */
+export function encodeRFC3986(str: string): string {
+  return encodeURIComponent(str).replace(
+    /[!'()*]/g,
+    (x) =>
+      `%${x
+        .charCodeAt(0)
+        .toString(16)
+        .toUpperCase()}`
+  );
+}
+
+/**
+ * Escapes all of the valid RegExp characters of a string.
+ * @param str The string that needs to be escaped.
+ */
 export function escapeRegExp(str: string): string {
-  return encodeURIComponent(str)
-    .replace(
-      /[!'()*]/g,
-      (x) =>
-        `%${x
-          .charCodeAt(0)
-          .toString(16)
-          .toUpperCase()}`
-    ) // RFC 3986.
-    .replace(/[-[\]{}()*+?.,\\^$|#\s]/g, "\\$&"); // All the RegExp sensitive characters.
+  return str.replace(/[-[\]{}()*+?.,\\^$|#\s]/g, "\\$&");
+}
+
+export type ReplacementDictionary = { [x: string]: string };
+
+/**
+ * Looks for the environment variables based on the keys of the given dictionary,
+ * then replaces the values found with each value from the same dictionary.
+ * @param replacements A dictionary of string keys and string values.
+ * @param content The content that has the text to be replaced.
+ */
+export function applyReplacementDictionary(
+  env: { [x: string]: string },
```

Comment:
can use your `ReplacementDictionary` type here too, right?

Reply:
I do, but should I? I wonder if @types/node has a specific type for process.env? I wish I could reuse that.

Reply:
I think this utility should be generic, and process.env should be structurally compatible with it, rather than this method being strongly tied to process.env's exact type

Reply:
So, DefinitelyTyped uses NodeJS.ProcessEnv. I'll use that one. https://github.com/DefinitelyTyped/DefinitelyTyped/blob/a277fa20b586433979d6b5dd61b6fac42cc724f1/types/node/v10/child_process.d.ts#L343